src/builder.js

```diff
@@ -30,12 +30,12 @@ class Builder {
   }
 
   async ensureAccountIsOnboarded(allowedRoles) {
-    const role = await this.rolesRepository.onboardedRole(getDefaultAddress(this.web3));
-    if (role.name === 'NONE') {
+    this.role = await this.rolesRepository.onboardedRole(getDefaultAddress(this.web3));
```

Comment:
Instead of `this.role` I would just return the role the node is onboarded for. Instance variables that are set or not depending on a call to a method is somewhat of an anti-pattern. I know we are doing it already with the build method. But this is not something we should duplicate:D